pkg/goDB/storage/gpfile/gpdir.go

```diff
@@ -532,6 +529,15 @@ func (d *GPDir) setPermissions(permissions fs.FileMode) {
 	d.permissions = permissions
 }
 
+// GenPathForTimestamp provides a unified generator method that allows to construct the path to
+// the data on disk based on a base path and a timestamp
+func GenPathForTimestamp(basePath string, timestamp int64) string {
```

Comment:
Exposed this so that _the_ way of generating the destination directory is globally available (to avoid some client or whatever trying to replicate the generation, potentially wreaking havoc).